R/checks.R

```diff
@@ -18,3 +18,91 @@
   x <- generation_time(10)
   checkmate::assert_numeric(x, len = 10)
 }
+
+#' Check inputs to `simulate_chains()` and `simulate_summary()`
+#'
+#' @param sim_func <character>; The simulation function to check
+#'
+#' @inheritParams simulate_chains
+#' @return NULL; called for side effects
+#' @keywords internal
+.check_sim_args <- function(
+    func_name = c("simulate_chains", "simulate_summary"),
+    index_cases,
+    statistic,
+    offspring_dist,
+    stat_max,
+    pop,
+    percent_immune,
+    tf_specified,
+    tf = NULL,
+    generation_time = NULL,
+    t0 = NULL) {
+  # Get the function name
+  func_name <- match.arg(func_name)
+  # Input checking
+  checkmate::assert_count(
+    index_cases,
+    positive = TRUE
+  )
+  # check that offspring is a function with argument "n"
+  .check_offspring_func_valid(offspring_dist)
+  # check that arguments related to the statistic are valid
+  .check_statistic_args(
+    statistic,
+    stat_max
+  )
+  checkmate::assert(
+    is.infinite(pop) ||
+      checkmate::assert_integerish(pop, lower = 1)
+  )
+  checkmate::assert_number(
+    percent_immune,
+    lower = 0, upper = 1
+  )
+
+  if (func_name == "simulate_chains") {
+    if (!is.null(generation_time)) {
+      .check_generation_time_valid(generation_time)
+    } else if (tf_specified) {
+      stop("If `tf` is specified, `generation_time` must be specified too.")
+    }
+    checkmate::assert_number(
+      tf,
+      lower = 0
+    )
+    checkmate::assert_numeric(
+      t0,
+      lower = 0, finite = TRUE
+    )
+  }
```

Comment:
Sorry I didn't think of this earlier but: If we made this snippet a separate function `.check_time_args` then we could do away with passing the function name (which I found really hard to parse, but that may of course be just me). What do you think?

Reply:
Ah, yes! That's a good idea. Thanks. I'm working on that now.

Reply:
I've now made the change here 8bd77e8aac80fba17786a0ecc11f83483aac57df.